### ISSUE TITLE
generate test result for each epoch as well

### ISSUE THREAD

**Issue by smilesun** [compstat-lmu/paper_2019_variationalResampleDistributionShift#15]
now we have per epoch result for the train and validation set,  it would be nice to have it for test set as well and with a flag to disable that, we might want to draw 3 curves for each set along the training epoch
